mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -589,7 +608,7 @@ private class LeastSquaresAggregator(
  * It's used in Breeze's convex optimization routines.
  */
 private class LeastSquaresCostFun(
-    data: RDD[(Double, Vector)],
+    data: RDD[(Double, Vector, Double)],
```

Comment:
Refactor the `Instance` case class out from LoR, and use it for code readability.